src/utils.js

```diff
@@ -232,8 +287,96 @@ var nmPhoneUtils = {
    * @param  {Number} length
    * @return {String}
    */
-  shortenToLength: function (value, length) {
-    return value.substring(0, length);
+  shortenToLength: shortenToLength,
+
+  /**
+   * Format the phoneData object into a string of the form:
+   *
+   * +<countrycode>-[<carriercode>-]<number>
+   *
+   * @param  {Object} phoneData
+   * @return {String}
+   */
+  getFormattedNumber: function getFormattedNumber(phoneData) {
+    if (!phoneData.cellTokens) {
+      return '';
+    }
+
+    return [
+        (!!phoneData.cellTokens.countryCode ? '+' + phoneData.cellTokens.countryCode : ''),
+        (''+phoneData.cellTokens.carrierCode),
+        (''+phoneData.cellTokens.number),
+      ].filter(function(v) { return !_.isEmpty(v); }).join('-');
+  },
+
```

Comment:
I remember having some issue with this. I'll try again your way and see if it works :)